### ISSUE TITLE
Query ignores limit when pasted multiline query contains comments

### ISSUE THREAD

**Issue by jmsfltchr** [graknlabs/console#109]
## Description

When pasting a multi-line query into the console, where the query contains comment and a limit, the limit is ignored.

# Reproducible Steps

Add and commit a basic schema:
```
define 
person sub entity, 
  plays friendship:friend;
friendship sub relation, 
  relates friend;
```

Insert 5 instances of `friendship`:
```
insert $p1 isa person; $p2 isa person; (friend: $p1, friend: $p2) isa friendship;
```
Run this 5 times and commit.

Try making the following query. Typing it out into the console vs copy-pasting it into the console. In the latter case the limit is ignored!
```
match $f($p1, $p2) isa friendship;
$p1 isa person; # my comment
$p2 isa person;
limit 2;
```

To prove I'm not crazy (first is pasted, second is typed):
```
test::data::read> match $f($p1, $p2) isa friendship;
> $p1 isa person; # my comment
> $p2 isa person;
> limit 2;

{ p1 iid 0x966e80018000000000000000 isa person; p2 iid 0x966e80018000000000000001 isa person; f iid 0xaa8280018000000000000003 (friend: iid 0x966e80018000000000000001, friend: iid 0x966e80018000000000000000) isa friendship; }
{ p1 iid 0x966e80018000000000000002 isa person; p2 iid 0x966e80018000000000000003 isa person; f iid 0xaa8280018000000000000004 (friend: iid 0x966e80018000000000000003, friend: iid 0x966e80018000000000000002) isa friendship; }
{ p1 iid 0x966e80018000000000000009 isa person; p2 iid 0x966e80018000000000000004 isa person; f iid 0xaa8280018000000000000001 (friend: iid 0x966e80018000000000000004, friend: iid 0x966e80018000000000000009) isa friendship; }
{ p1 iid 0x966e80018000000000000006 isa person; p2 iid 0x966e80018000000000000007 isa person; f iid 0xaa8280018000000000000002 (friend: iid 0x966e80018000000000000007, friend: iid 0x966e80018000000000000006) isa friendship; }
{ p1 iid 0x966e80018000000000000008 isa person; p2 iid 0x966e80018000000000000005 isa person; f iid 0xaa8280018000000000000000 (friend: iid 0x966e80018000000000000005, friend: iid 0x966e80018000000000000008) isa friendship; }
{ p1 iid 0x966e80018000000000000003 isa person; p2 iid 0x966e80018000000000000002 isa person; f iid 0xaa8280018000000000000004 (friend: iid 0x966e80018000000000000003, friend: iid 0x966e80018000000000000002) isa friendship; }
{ p1 iid 0x966e80018000000000000004 isa person; p2 iid 0x966e80018000000000000009 isa person; f iid 0xaa8280018000000000000001 (friend: iid 0x966e80018000000000000004, friend: iid 0x966e80018000000000000009) isa friendship; }
{ p1 iid 0x966e80018000000000000001 isa person; p2 iid 0x966e80018000000000000000 isa person; f iid 0xaa8280018000000000000003 (friend: iid 0x966e80018000000000000001, friend: iid 0x966e80018000000000000000) isa friendship; }
{ p1 iid 0x966e80018000000000000005 isa person; p2 iid 0x966e80018000000000000008 isa person; f iid 0xaa8280018000000000000000 (friend: iid 0x966e80018000000000000005, friend: iid 0x966e80018000000000000008) isa friendship; }
{ p1 iid 0x966e80018000000000000007 isa person; p2 iid 0x966e80018000000000000006 isa person; f iid 0xaa8280018000000000000002 (friend: iid 0x966e80018000000000000007, friend: iid 0x966e80018000000000000006) isa friendship; }
test::data::read> match $f($p1, $p2) isa friendship;
                  $p1 isa person; # my comment
                  $p2 isa person;
                  limit 2;

{ p1 iid 0x966e80018000000000000006 isa person; p2 iid 0x966e80018000000000000007 isa person; f iid 0xaa8280018000000000000002 (friend: iid 0x966e80018000000000000007, friend: iid 0x966e80018000000000000006) isa friendship; }
{ p1 iid 0x966e80018000000000000009 isa person; p2 iid 0x966e80018000000000000004 isa person; f iid 0xaa8280018000000000000001 (friend: iid 0x966e80018000000000000004, friend: iid 0x966e80018000000000000009) isa friendship; }
test::data::read>
```

**Comment by haikalpribadi** [comment]
Is it only the `limit` that's ignored, or are the last 2 lines that get ignored, @jmsfltchr ?

**Comment by jmsfltchr** [comment]
Very good question! Looks like you're idea is right there, I tried rearranging the query:
```
match 
$p1 isa person; 
$p2 isa person; # my comment
$f($p1, $p2) isa friendship;
limit 2;
```
And the results I see only contain `$p1` and `$p2` and not `$f`, implying that all lines after the comment are being ignored.

**Comment by haikalpribadi** [comment]
I think the parser in JLine might be the culprit

**Comment by vmax** [comment]
I've attempted reproducing it in `2.0.0-alpha` (bug **is** present) and on latest `master` (bug **is not** present), therefore the fix is somewhere in between. So I did test every commit in `git log --oneline 2.0.0-alpha...master` and it occurs that this bug is fixed in #105.
